modules/permission-sets/variables.tf

```diff
@@ -1,16 +1,16 @@
 variable "permission_sets" {
   type = list(object({
     name               = string
-    description        = string
-    relay_state        = string
-    session_duration   = string
-    tags               = map(string)
-    inline_policy      = string
-    policy_attachments = list(string)
-    customer_managed_policy_attachments = list(object({
+    description        = optional(string, "")
+    relay_state        = optional(string, "")
+    session_duration   = optional(string, "PT8H")
+    tags               = optional(map(string), {})
+    inline_policy      = optional(string, "")
+    policy_attachments = optional(list(string), [])
+    customer_managed_policy_attachments = optional(list(object({
```

Comment:
This worries me as this sub-module is all about making PermissionSets and yet you're making it so everything is optional at the moment, which then seems to defeat the purpose of defining PermissionSets? Mind expanding upon the purpose of making everything optional? For example, we don't want `description` to be optional. Everytime someone creates a PS they should be forced to write a description for it as that will help the next person understand why that PS was created and what it was used for.